etc/docker-scripts/bootstrapper.py

```diff
@@ -18,243 +18,334 @@
+        """
+        return os.path.join(self.input_params.get('cos-directory', ''), filename)
+
+    def get_file_from_object_storage(self, file_to_get: str) -> None:
+        """
+        Utility function to get files from an object storage
+
+        :param file_to_get: filename
+        """
+
+        print('Get file {} from bucket {}'.format(file_to_get, self.cos_bucket))
+        object_to_get = self.get_object_storage_filename(file_to_get)
+
+        self.cos_client.fget_object(bucket_name=self.cos_bucket,
+                                    object_name=object_to_get,
+                                    file_path=file_to_get)
+
+    def put_file_to_object_storage(self, file_to_upload: str, object_name: Optional[str] = None) -> None:
+        """
+        Utility function to put files into an object storage
+        :param file_to_upload: filename
+        :param object_name: remote filename (used to rename)
+        """
+
+        object_to_upload = object_name
+        if not object_to_upload:
+            object_to_upload = file_to_upload
+
+        print('Uploading file {} as {} to bucket {}'.format(file_to_upload, object_to_upload, self.cos_bucket))
+
+        object_to_upload = self.get_object_storage_filename(object_to_upload)
+
+        self.cos_client.fput_object(bucket_name=self.cos_bucket,
+                                    object_name=object_to_upload,
+                                    file_path=file_to_upload)
+
+    def has_wildcard(self, filename):
+        wildcards = ['*', '?']
+        return bool(any(c in filename for c in wildcards))
+
+    def process_output_file(self, output_file):
+        """Puts the file to object storage.  Handles wildcards and directories. """
+
+        matched_files = [output_file]
+        if self.has_wildcard(output_file):  # explode the wildcarded file
+            matched_files = glob.glob(output_file)
+
+        for matched_file in matched_files:
+            if os.path.isdir(matched_file):
+                for file in os.listdir(matched_file):
+                    self.process_output_file(os.path.join(matched_file, file))
+            else:
+                self.put_file_to_object_storage(matched_file)
+
+
+class NotebookFileOp(FileOpBase):
+    """Perform Notebook File Operation"""
+
+    def execute(self) -> None:
+        """Execute the Notebook and upload results to object storage"""
+        notebook = os.path.basename(self.filepath)
+        notebook_name = notebook.replace('.ipynb', '')
+        notebook_output = notebook_name + '-output.ipynb'
+        notebook_html = notebook_name + '.html'
+
+        print("Executing notebook through Papermill: {} ==> {}".format(notebook, notebook_output))
+        try:
+            import papermill
+            papermill.execute_notebook(
+                notebook,
+                notebook_output,
+                kernel_name="python3"
+                # parameters=
+            )
```

Comment:
The following change should resolve elyra #845 - using the kernel from within the notebook file.
```suggestion
           subprocess.run(['papermill', notebook, notebook_output], check=True)
```